### ISSUE TITLE
Wrong x offset for view controller

### ISSUE THREAD

**Issue by nplkhn** [rechsteiner/Parchment#575]
I have an issue then i rotate screen the x offset calculates with mistake and view controller from the next page appears on the second half of the first page.

My paging view controller embedded in a navigation controller. That navigation controller added as a child view controller to other view controller that also embedded in navigation view controller.

If I have only one navigation controller everything works good.

**Comment by Starsky89** [comment]
I had something similar, when I would rotate the iPad, and the content of the underlying scrollView wasn't updated correctly.
Here is the solution which worked for me:

```
override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
        super.viewWillTransition(to: size, with: coordinator)
        
        paging.pageViewController.scrollView.invalidateIntrinsicContentSize()
    }
```

`paging` is of type `PagingViewController`, part of Parchment.

**Comment by Starsky89** [comment]
I found the solution to this problem:
Call this function inside the subclassed object of type `PagingViewController`:
```
    override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
        super.viewWillTransition(to: size, with: coordinator)
        
        self.pageViewController.willTransition(to: view.traitCollection, with: coordinator)
    }
    ```
    
    @rechsteiner 
    Can I ask you if this is intentional, or was it a mistake.
    In the file `PageViewController.swift`, there is this override method:
    ```
    public override func willTransition(to newCollection: UITraitCollection, with coordinator: UIViewControllerTransitionCoordinator) {
        super.willTransition(to: newCollection, with: coordinator)
        coordinator.animate(alongsideTransition: { _ in
            self.manager.viewWillTransitionSize()
        })
    }
   ```
   I wonder if this should have been an override for this function?
   ```
   public override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
        super.viewWillTransition(to: size, with: coordinator)
        
        coordinator.animate(alongsideTransition: { _ in
            self.manager.viewWillTransitionSize()
        })
    }
    ```
    
    This would solve the problem we're having on iPads when rotating the device, the underlying scrollView's content width should be updated.

**Comment by rechsteiner** [comment]
Ahh yes, that definitely looks like a mistake. Will get that fixed soon!

**Comment by rechsteiner** [comment]
I've fixed some issues with rotation when using the SwiftUI wrapper in #596. Could you target the master branch and see if that fixes your issues?

**Comment by Starsky89** [comment]
@rechsteiner 
Yeah, it looks like this fixes the problem with rotation. 

P.S: You might want to remove the print statement on line 171
`print("viewWillTransition: ", size)` inside `PageViewController.swift`

**Comment by rechsteiner** [comment]
A new version is out now that fixes this `v3.1.0`. @Starsky89 I forgot to remove the print statement in this release, but it's fixed in the master branch
